src/liboslexec/pointcloud.cpp

```diff
@@ -366,8 +369,38 @@ RendererServices::pointcloud_get (ShaderGlobals *sg,
     // then copy them back to the caller's indices.
 
     // Actual data query
-    const_cast<Partio::ParticlesData *>(cloud)->data (*attr, count, (Partio::ParticleIndex *)indices,
-                 true, out_data);
+    if (partio_type == OIIO::TypeString) {
+        // strings are special cases because they are stored as int index
+        // We don't know how many, but the usual case is just one. So use
+        // a local var for the usual case, but be prepared to use heap.
+        // FIXME: this would be a great spot for a "short vector.""
+        std::unique_ptr<int[]> strindices_alloc;  // in case we need > 1
+        int strindices_local;
+        int* strindices = &strindices_local;
+        if (count > 1) {
+            strindices_alloc.reset(new int[count]);
+            strindices = strindices_alloc.get();
+        }
+        const_cast<Partio::ParticlesData*>(cloud)->data (*attr, count,
+                    (const Partio::ParticleIndex *)indices, true, (void*)strindices);
+        const auto& strings = cloud->indexedStrs(*attr);
+        int sicount = int(strings.size());
+        for (size_t i = 0; i < count; ++i) {
+            int ind = strindices[i];
+            if (ind >= 0 && ind < sicount)
+                ((ustring *)out_data)[i] = ustring(strings[ind]);
+            else
+                ((ustring *)out_data)[i] = ustring();
+        }
+    } else {
+        // All cases aside from strings are simple.
+        const_cast<Partio::ParticlesData*>(cloud)->data (*attr, count,
+                    (const Partio::ParticleIndex *)indices, true, out_data);
+        // FIXME: it is regrettable that we need this const_cast (and the
```

Comment:
I believe this is fixed already but we are on an old version?

Reply:
Maybe? We are on a... customized version. We need to submit our changes back to the project and then rebase.

Reply:
But in fact, this is not fixed in its current master: https://github.com/wdas/partio/blob/master/src/lib/Partio.h#L115

Reply:
Fair enough!

Reply:
All data() does it call dataInternalMultiple(), which is const. But data() is not marked as const. Oh well.